oidc_middleware.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+)
+// Largely taken from https://developer.okta.com/blog/2021/01/04/offline-jwt-validation-with-go
+
+func GetOidcPublicKeys(config OICDConfig) *map[string]*rsa.PublicKey {
+	rsakeys := make(map[string]*rsa.PublicKey)
+	var body map[string]interface{}
+	resp, err := http.Get(config.KeysURL)
+	if err != nil {
+		log.Error().WithError(err).Message("Could not fetch from KeysURL.")
+	}
```

Comment:
Missing return statement after the log.

Suggest moving the logging to the caller, and only return errors from this function instead.

Reply:
These are steup errors - they probably should invoke panics.

Reply:
No need for panics when you can return error here and then let the caller end the program safely instead

Reply:
Im not arguing we need to invoke panic. Im arguing we should invoke panic. The reason being that further execution would be unsafe. If the panic gets invoke here we could tell the user to fix the OIDC config such that it works, or alternativly disable the OIDC config such that it runs unsafe.

Reply:
Resolved in pairprogramming session